metricflow/test/model/validations/test_materializations.py

```diff
@@ -0,0 +1,280 @@
+    assert (
+        len(
+            ValidMaterializationRule.validate_model(
+                model_with_materialization(
+                    simple_model__pre_transforms,
+                    [
+                        Materialization(
+                            name="foobar",
+                            metrics=["bookings"],
+                            dimensions=["ds", "is_instant", "listing__country_latest"],
+                            destinations=[
+                                MaterializationDestination(
+                                    location=MaterializationLocation.DW,
+                                    format=MaterializationFormat.WIDE,
+                                    rollups=[["ds", "invalid_dimension"]],
+                                )
+                            ],
+                        )
+                    ],
+                )
+            )
+        )
+        == 1
+    )
+
+
+def test_valid_time_granularity(simple_model__pre_transforms: UserConfiguredModel) -> None:  # noqa: D
+    assert (
+        len(
+            ValidMaterializationRule.validate_model(
+                model_with_materialization(
+                    simple_model__pre_transforms,
+                    [
+                        Materialization(
+                            name="materialization_test_case",
+                            metrics=["bookings"],
+                            dimensions=["ds__day"],
+                        )
+                    ],
+                )
+            )
+        )
+        == 0
+    )
+
+
+def test_invalid_time_granularity(simple_model__pre_transforms: UserConfiguredModel) -> None:  # noqa: D
+    assert (
+        len(
+            ValidMaterializationRule.validate_model(
+                model_with_materialization(
+                    simple_model__pre_transforms,
+                    [
+                        Materialization(
+                            name="materialization_test_case",
+                            metrics=["revenue"],
+                            dimensions=["ds__hour"],
+                        )
+                    ],
+                )
+            )
+        )
+        == 2
+    )
+
+
+def test_missing_primary_time_dimension_in_rollups(
+    simple_model__pre_transforms: UserConfiguredModel,
+) -> None:  # noqa: D
+    """Test to ensure the primary time dimension is used in materialization destination"""
+    model = copy.deepcopy(simple_model__pre_transforms)
+    func_for_materialization: Callable[[Materialization], bool] = (
+        lambda materialization: len(materialization.destinations) > 0
+    )
+    materialization_with_destinations, _ = find_materialization_with(model, func_for_materialization)
+    assert type(materialization_with_destinations) == Materialization
+
+    # We assume the primary time dimension is `ds`
+    primary_time = "ds"
```

Comment:
So I get it, but this should probably be fixed at some point. At the very least, update the comment to reflect that the primary time dimension in the simple_model is, in fact, called "ds"